### ISSUE TITLE
[REGRESSION]: certain child process commands hangs in hooks

### ISSUE THREAD

**Issue by haleksandre** [microsoft/playwright#26426]
**Context:**
- GOOD Playwright Version: 1.31
- BAD Playwright Version: 1.36
- Operating System: Linux
- Extra: Podman container

**Code Snippet**

```javascript
...
test.beforeEach(async ({ page }) => {
	const { execa } = await import("execa");

 	// this works
	const version = await execa("node", ["--version"]);
	console.log(version.stdout);
	console.error(version.stderr);

	// this hangs & trigger the timedout
	const file = await execa("node", ["tests/version.js"]); // tests/version.js > console.log(process.version)
	console.log(file.stdout);
	console.error(file.stderr);

	await page.goto("https://demo.playwright.dev/todomvc");
});
...
```

**Describe the bug**

When executing a child process within a fixture/hook some command work as expected, but some hang until the hook hit the timeout. I've made the code snippet available within the [reproduction repo](https://github.com/haleksandre/playwright).

**Comment by Shasvinth** [comment]
Try this code
```
import { defineConfig, devices } from '@playwright/test';
import path from 'path';
import execa from 'execa';

// ... (other configuration settings)

export default defineConfig({
  // ... (other configuration settings)

  testDir: './tests/e2e',
  use: {
    // ... (other configuration settings)

    // Modify the webServer settings if needed
    webServer: {
      command: 'npm run dev',
      url: 'http://localhost:4000/',
      reuseExistingServer: true,
    },
  },

  // ... (other configuration settings)

  // Define your test fixtures or hooks
  fixtures: [/* ... */],

  // Define your test cases
  tests: [

    test.beforeEach(async ({ page }) => {
      // Use try-catch to handle errors
      try {
        const { stdout, stderr } = await execa("node", ["--version"]);
        console.log(stdout);
        console.error(stderr);

        const file = await execa("node", ["tests/version.js"]);
        console.log(file.stdout);
        console.error(file.stderr);
      } catch (error) {
        console.error("Error executing child process:", error);
      }

      await page.goto("https://demo.playwright.dev/todomvc");
    }),

  
  ],
});

```

**Comment by haleksandre** [comment]
In my project, the execa commands are wrapped in a `try catch` like your example, but the commands aren't failing so they aren't triggering the `catch` block. They are just hanging until the hook timeout kicks in. I've tried to dig into what is hanging for certain commands & it seems to be these promises [here](https://github.com/sindresorhus/execa/blob/468fbe15edd3f2ec74b878be609ba2a950c4068e/lib/stream.js#L110) never gets resolved. If I run the file or the commands outside the hook via `node test.js` everything is working fine.

```ts
// test.js
import { execa } from "execa";

const version = await execa("node", ["--version"]);
console.log(version.stdout);
console.error(version.stderr);

const file = await execa("node", ["tests/version.js"]); // tests/version.js > console.log(process.version)
console.log(file.stdout);
console.error(file.stderr);
```

The execa commands within the hook were working fine before I updated the latest version. So my guess is that there's something within the hook worker that sometimes conflicts with the child process execa spawn which somehow makes the stdout/stderr promises hang.

Maybe this could help, for my use case, I use execa with prisma to migrate & seed a new database before certain tests & `prisma db push` as well as my project's seed command `tsx cli/cli.ts seed` commands hangs.
